### PR TITLE
Introduce apprentice limit to disable lessons

### LINKED ISSUE #363: Lesson Limit
Prevent further lessons from being completed over a certain Apprentice count to help prevent burnout.

### PR DESCRIPTION
**Pull request by ShaneCoufreur**
Attempted a fix #363

### CODE REVIEW

**Comment by UInt2048** [comment]
@ShaneCoufreur Haven't had time to review this, but if you remove the word "for" it will auto-close the issue if merged

File: ios/SettingsViewController.swift [hunk 1]
```diff
@@ -259,6 +267,12 @@ class SettingsViewController: UITableViewController {
     "\(Settings.lessonBatchSize)"
   }
 
+  private var apprenticeLessonsLimitText: String {
+    Settings.apprenticeLessonsLimit != Int.max
+      ? "\(Settings.apprenticeLessonsLimit)"
+      : "Disabled"
```
Comment (davidsansome) [inline ios/SettingsViewController.swift:273]: I wonder if "Disabled" sounds a bit technical. Maybe just "-" or "None" would work better?

File: ios/TKMLessonApprenticeLimitViewController.m [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 David Sansome
```
Comment (davidsansome) [inline ios/TKMLessonApprenticeLimitViewController.m:1]: Would you be able to rewrite this in Swift instead? I'm trying not to introduce any new Objective-C

**Review by davidsansome — CHANGES_REQUESTED** [review]

File: ios/TKMLessonApprenticeLimitViewController.m (hunk 2 above)
Comment (ShaneCoufreur) [reply ios/TKMLessonApprenticeLimitViewController.m:1]: I'll see if I can brush up on some Swift soon. Don't have a whole lot of experience with it, but will see what I can do.

**Comment by UInt2048** [comment]
I rewrote the Objective-C in Swift - see #450

**Pull request closed without merging**